sdk/notificationhubs/notification-hubs/review/notification-hubs.api.md

```diff
@@ -0,0 +1,177 @@
+    platform: "baidu";
+}
+
+// @public
+export interface BrowserInstallation extends Installation {
+    platform: "browser";
+    pushChannel: BrowserPushChannel;
+}
+
+// @public
+export interface BrowserMessage extends JSONNotificationMessage {
+    platform: "browser";
+}
+
+// @public
+export interface BrowserPushChannel {
+    auth: string;
+    endpoint: string;
+    p256dh: string;
+}
+
+// @public
+export interface DeviceTokenInstallation extends Installation {
+    pushChannel: string;
+}
+
+// @public
+export interface FirebaseLegacyInstallation extends DeviceTokenInstallation {
+    platform: "gcm";
+}
+
+// @public
+export interface FirebaseLegacyMessage extends JSONNotificationMessage {
+    platform: "gcm";
+}
+
+// @public
+export interface Installation {
+    readonly expirationTime: string;
+    installationId: string;
+    readonly lastUpdate: string;
+    platform: string;
+    tags?: string[];
+    templates?: Record<string, InstallationTemplate>;
+    userId?: string;
+}
+
+// @public
+export interface InstallationPatch {
+    op: JSONPatchType;
+    path: string;
+    value?: string;
+}
+
+// @public
+export interface InstallationTemplate {
+    body: string;
+    expiry?: string;
+    headers: Record<string, string>;
+    tags?: string[];
+}
+
+// @public
+export type InstallationType = AppleInstallation | ADMInstallation | BaiduInstallation | BrowserInstallation | FirebaseLegacyInstallation | WindowsInstallation;
+
+// @public
+export interface JSONNotificationMessage extends NotificationHubMessage {
+    contentType: "application/json;charset=utf-8";
+}
+
+// @public
+export type JSONPatchType = "add" | "remove" | "replace";
+
+// @public
+export interface NotificationHubMessage {
+    body: string;
+    contentType: string;
+    headers: Record<string, string>;
+    platform: string;
+}
+
+// @public
+export type NotificationHubMessageType = AppleMessage | ADMMessage | BaiduMessage | BrowserMessage | FirebaseLegacyMessage | WindowsMessage | TemplateMessage;
+
+// @public
+export interface NotificationHubResponse {
+    correlationId?: string;
+    location?: string;
+    trackingId?: string;
+}
+
+// @public
+export class NotificationHubsClient extends ServiceClient {
+    constructor(connectionString: string, hubName: string, options?: NotificationHubsClientOptions);
+    deleteInstallation(installationId: string, options?: OperationOptions): Promise<NotificationHubResponse>;
+    getFeedbackContainerURL(options?: OperationOptions): Promise<URL>;
+    getInstallation(installationId: string, options?: OperationOptions): Promise<InstallationType>;
+    patchInstallation(installationId: string, installationPatches: InstallationPatch[], options?: OperationOptions): Promise<NotificationHubResponse>;
+    scheduleNotification(scheduledTime: Date, tags: string[] | string, message: NotificationHubMessageType, options?: SendOperationOptions): Promise<NotificationHubResponse>;
```

Comment:
in the past we've avoided doing this and just taken an array always, since it's not hard to type `[]` - I don't feel strongly if single tags are incredibly common though

Reply:
An array doesn't cover where the tags is an expression of `oneTag&&twoTag&&threeTag` versus the array which is then turned into `oneTag||twoTag||threeTag` so there's a big difference.